code/__HELPERS/roundend.dm

```diff
@@ -583,12 +583,35 @@
 		Trigger()
 		return
 
+///Returns a custom title for the roundend credit/report
+/proc/get_custom_title_from_id(obj/item/card/id/I, mind_job, mind_name, newline=FALSE)
```

Comment:
make a wrapper that accepts a mob and gets this all off of it if applicable, and then use it in the roundend code

Reply:
I am not sure if it's what you'd like to see, but changed now.

Reply:
works